doc/dds-core-xtypes/README.md

```diff
@@ -235,51 +236,53 @@ node.access().member() // member used for access from parent to this node
 
 ## Performance
-The `DynamicData` instance only uses the minimal allocations needed to store the data.
+The `DynamicData` instance uses the minimal allocations needed to store any data.
 If its associated type only contains the following (nested) types: `PrimitiveType`, `StructType`, or `ArrayType`;
-the memory required can be got with only **one allocation** at the creation phase of the `DynamicData`.
-Only if the type contains some `MutableCollectionType`, another allocation will be performed to manage the inner types of that mutable type.
+the memory required can be allocated with only **one allocation** during the creation phase of the `DynamicData`.
+On the other hand, if the type contains any `MutableCollectionType`, other allocations will be performed in order to manage that inner types' mutable members.
 
 Let see an example:
 ```c++
-StructType inner("Inner");
+StructType inner("inner");
 inner.add_member("m_int", primitive_type<int>());
 inner.add_member("m_float", primitive_type<float>());
 inner.add_member("m_array", ArrayType(primitive_type<uint16_t>(), 4));
 
 DynamicData data(inner);
 ```
-This data from `inner` will be represented in memory as follow, with only one allocation:
+`DynamicData data(inner)` will be represented in memory as follows. Only one memory allocation is needed:
 ![](inner-memory.png)
 
 The next complex type:
 ```c++
-StructType outter("Outter");
-outter.add_member("m_inner", inner);
-outter.add_member("m_array_inner", ArrayType(inner, 4));
-outter.add_member("m_sequence_inner", SequenceType(inner, 4));
-outter.add_member("m_string", StringType());
+StructType outer("outer");
+outer.add_member("m_inner", inner);
+outer.add_member("m_array_inner", ArrayType(inner, 4));
+outer.add_member("m_sequence_inner", SequenceType(inner, 4));
+outer.add_member("m_string", StringType());
 
-DynamicData data(outter);
+DynamicData data(outer);
 ```
-Needs two more allocations, one for the secuence, and another one for the string:
-![](outter-memory.png)
+In this case, two allocations will be needed: one for the sequence, and a second one for the string:
+![](outer-memory.png)
 
 ## Debugging DynamicData
-As a `DynamicData` is totally built in runtime, no static checks can be done to ensure the correct behaviour.
-As an attempt to solve this, many methods are checked with `asserts`, to avoid the overload of checks in *release mode*.
-We strongly recommend to compile in *debug mode* during developing phase to allow `xtypes` library to perform all possible checks.
-In the same way, to improve the performance, compile in *release mode* for production, in order to avoid the cost of the checks.
+As a `DynamicData` is fully built at runtime, no static checks can ensure its correct behaviour.
+As an attempt to solve this, asserts have been implemented accross various methods to avoid the overload of checks in *release mode*.
```

Comment:
```suggestion
As an attempt to solve this, asserts have been placed accross various methods to avoid the overload of checks in *release mode*.
```
We dont implement the asserts, only place them in some location.